pkg/infoschema/builder.go

```diff
@@ -896,9 +900,9 @@ type tableFromMetaFunc func(alloc autoid.Allocators, tblInfo *model.TableInfo) (
 func (b *Builder) createSchemaTablesForDB(di *model.DBInfo, tableFromMeta tableFromMetaFunc, schemaVersion int64) error {
 	schTbls := &schemaTables{
 		dbInfo: di,
-		tables: make(map[string]table.Table, len(di.Tables)),
+		tables: make(map[string]table.Table, len(di.Tables())),
 	}
-	for _, t := range di.Tables {
+	for _, t := range di.Tables() {
```

Comment:
If we switch to SchemaTables we also don't need SetTables in fetchSchemasWithTables